src/hooks/siteDashboardHooks/useUpdateViewedReviewRequests.ts

```diff
@@ -0,0 +1,17 @@
+import type { AxiosError } from "axios"
+import { useMutation } from "react-query"
+import type { UseMutationResult } from "react-query"
+
+import * as SiteDashboardService from "services/SiteDashboardService"
+
+export const useUpdateViewedReviewRequests = (): UseMutationResult<
+  void,
+  AxiosError<{ message: string }>,
+  { siteName: string }
+> => {
+  return useMutation<
+    void,
+    AxiosError<{ message: string }>,
+    { siteName: string }
+  >(({ siteName }) => SiteDashboardService.updateViewedReviewRequests(siteName))
```

Comment:
hmm why don't we invalidate the cache for the related getter?

Reply:
could i clarify the difference between this and the above hook? is it just a matter of scale? (one above is single rr, this is for all rr)

Reply:
> hmm why don't we invalidate the cache for the related getter?

I am thinking that the review request should always be shown as new to the user, until the user explicitly refreshes the page or navigates to another page and comes back. I'm not sure if my understanding of the react query cache is correct here, but I think if I were to invalidate the cache here, it would cause the review request to no longer show as new even though the user did not perform any page refreshes, which may be too fast for the user to notice.

> could i clarify the difference between this and the above hook? is it just a matter of scale? (one above is single rr, this is for all rr)

Yup that interpretation is correct.

Reply:
ah ok that explanation makes sense! thanks for clarifying!